File: core/kora-app-annotation-processor/src/main/java/io/koraframework/kora/app/annotation/processor/ResolvedGraph.java
package io.koraframework.kora.app.annotation.processor;

import com.palantir.javapoet.ClassName;
import io.koraframework.kora.app.annotation.processor.component.ResolvedComponent;

import javax.lang.model.element.TypeElement;
import java.util.List;
import java.util.Map;

public record ResolvedGraph(TypeElement root, List<TypeElement> allModules, List<ResolvedComponent> components, Map<ClassName, ResolvedComponent> conditionByTag) {}

File: kora-app-symbol-processor/src/main/kotlin/io/koraframework/kora/app/ksp/ResolvedGraph.kt
package io.koraframework.kora.app.ksp

import com.google.devtools.ksp.symbol.KSClassDeclaration
import com.squareup.kotlinpoet.ClassName
import io.koraframework.kora.app.ksp.component.ResolvedComponent

class ResolvedGraph(
    val root: KSClassDeclaration,
    val allModules: List<KSClassDeclaration>,
    val components: List<ResolvedComponent>,
    val conditionByTag: MutableMap<ClassName, ResolvedComponent>
) {}
